src/hooks/wallets/mpc/useMPC.ts

```diff
@@ -0,0 +1,86 @@
+import { useCurrentChain } from '@/hooks/useChains'
+import { getRpcServiceUrl } from '../web3'
+import useOnboard, { connectWallet, getConnectedWallet } from '@/hooks/wallets/useOnboard'
+import { ONBOARD_MPC_MODULE_LABEL } from '@/services/mpc/module'
+
+const { getStore, setStore, useStore } = new ExternalStore<Web3AuthMPCCoreKit>()
+
+export const useInitMPC = () => {
+  const chain = useCurrentChain()
+  const onboard = useOnboard()
+
+  useEffect(() => {
+    if (!chain || !onboard) {
+      return
+    }
+    const chainConfig = {
+      chainId: `0x${Number(chain.chainId).toString(16)}`,
+      chainNamespace: CHAIN_NAMESPACES.EIP155,
+      rpcTarget: getRpcServiceUrl(chain.rpcUri),
+      displayName: chain.chainName,
+      blockExplorer: chain.blockExplorerUriTemplate.address,
+      ticker: chain.nativeCurrency.symbol,
+      tickerName: chain.nativeCurrency.name,
+    }
+
+    const currentInstance = getStore()
+    let previousChainChangedListeners: Function[] = []
+    if (currentInstance?.provider) {
+      // We are already connected. We copy onboards event listener for the chainChanged event to propagate a potentially new chainId
+      const oldProvider = currentInstance.provider
+      previousChainChangedListeners = oldProvider.listeners('chainChanged')
+    }
+
+    const web3AuthCoreKit = new Web3AuthMPCCoreKit({
+      web3AuthClientId: WEB3_AUTH_CLIENT_ID,
+      // Available networks are "sapphire_devnet", "sapphire_mainnet"
+      web3AuthNetwork: WEB3AUTH_NETWORK.DEVNET,
+      baseUrl: `${window.location.origin}/serviceworker`,
+      uxMode: 'popup',
+      enableLogging: true,
+      chainConfig,
+    })
+
+    web3AuthCoreKit
+      .init()
+      .then(() => {
+        setStore(web3AuthCoreKit)
+        // If rehydration was successful, connect to onboard
+        if (web3AuthCoreKit.status !== COREKIT_STATUS.LOGGED_IN || !web3AuthCoreKit.provider) {
+          return
+        }
+        const connectedWallet = getConnectedWallet(onboard.state.get().wallets)
+        if (!connectedWallet) {
+          connectWallet(onboard, {
+            autoSelect: {
+              label: ONBOARD_MPC_MODULE_LABEL,
+              disableModals: true,
+            },
+          }).catch((reason) => console.error('Error connecting to MPC module:', reason))
+        } else {
+          const newProvider = web3AuthCoreKit.provider
+
+          // To propagate the changedChain we disconnect and connect
+          if (previousChainChangedListeners.length > 0 && newProvider) {
+            previousChainChangedListeners.forEach((previousListener) =>
+              newProvider.addListener('chainChanged', (...args: []) => previousListener(...args)),
+            )
+            newProvider.emit('chainChanged', `0x${Number(chainConfig.chainId).toString(16)}`)
+          }
+        }
+      })
+      .catch((error) => console.error(error))
+  }, [chain, onboard])
+}
+
+export const getMPCCoreKitInstance = getStore
```

Comment:
Nit: let's export this prefixed as per my above comment. I see that this is only used inside the onboard module, which I _think_ is alright but it shouldn't used within React.

```suggestion
export const _getMPCCoreKitInstance = getStore
```